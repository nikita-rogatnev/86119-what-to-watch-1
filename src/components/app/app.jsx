import React from "react";
import {Switch, Route} from "react-router-dom";

import Home from "../pages/home/home";
import Film from "../pages/film/film";
import Mylist from "../pages/mylist/mylist";

import Header from "../header/header";
import SignIn from "../sign-in/sign-in";
import Player from "../video-player/video-player";
import Footer from "../footer/footer";
import Heading from "../heading/heading";

import withPrivateRoute from "../../hocs/with-private-route/with-private-route";

const PrivateRoute = withPrivateRoute(Route);

class App extends React.PureComponent {
  render() {
    return (
      <Switch>
        <Route path="/" exact component={Home}/>
        <Route
          path="/login"
          render={() => (
            <div className="user-page">
              <Header title={`Sign in`}/>
              <SignIn/>
              <Footer/>
            </div>
          )}
        />
        <Route path="/player" component={Player}/>
        <Route path="/film/:id" exact component={Film}/>
        <PrivateRoute path="/mylist" exact component={Mylist}/>
        <Route render={() => <Heading title={`Error 404`}/>}/>
      </Switch>
    );
  }
}

export default App;
