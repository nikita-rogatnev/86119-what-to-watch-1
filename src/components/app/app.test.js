import React from 'react';
import renderer from 'react-test-renderer';
import {BrowserRouter} from "react-router-dom";

import {App} from './app';

import mockData from '../../mocks/mock-data';
import mockFilters from '../../mocks/mock-filters';

jest.mock(`../header/header`, () => `Header`);

it(`App renders correctly`, () => {
  const tree = renderer
    .create(<BrowserRouter><App
      data={mockData}
      dataFavorite={mockData}
      filters={mockFilters}
      currentFilter={`All genres`}
      changeCurrentFilter={() => {
      }}
      isAuthorizationRequired={false}
    /></BrowserRouter>).toJSON();

  expect(tree).toMatchSnapshot();
});
