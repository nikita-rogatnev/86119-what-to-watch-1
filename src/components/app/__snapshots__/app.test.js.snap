// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<Switch>
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(Home)",
        "type": [Function],
      }
    }
    exact={true}
    path="/"
  />
  <Route
    path="/login"
    render={[Function]}
  />
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(Film)",
        "type": [Function],
      }
    }
    exact={true}
    path="/film/:id"
  />
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(Film)",
        "type": [Function],
      }
    }
    path="/film/:id/details"
  />
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(Film)",
        "type": [Function],
      }
    }
    path="/film/:id/reviews"
  />
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(Review)",
        "type": [Function],
      }
    }
    exact={true}
    path="/film/:id/review"
  />
  <Route
    component={[Function]}
    path="/player"
  />
  <ConnectFunction
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(Mylist)",
        "type": [Function],
      }
    }
    exact={true}
    path="/mylist"
  />
  <Route
    render={[Function]}
  />
</Switch>
`;
